src/XMakeTasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1512,17 +1512,55 @@ Copyright (C) Microsoft Corporation. All rights reserved.
   -->
   <Target Name="_GetProjectReferenceTargetFrameworkProperties"
           Outputs="%(_MSBuildProjectReferenceExistent.Identity)">
-    <!-- As a mitigation for https://github.com/Microsoft/msbuild/issues/1276
-       .vcxproj is treated specially to avoid the cost of double-evaluating
-       for easily-identifiable-as-not-cross-targeting C++ projects. -->
+    <!--
+      Honor SkipGetTargetFrameworkProperties=true metadata on project references
+      to mean that the project reference is known not to target multiple frameworks
+      and the mechanism here for selecting the best one can be skipped as an optimization.
+
+      We give this treatment to .vcxproj by default since no .vcxproj can target more
+      than one framework.
+   -->
+   <ItemGroup>
+      <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SkipGetTargetFrameworkProperties)' == '' and '%(Extension)' == '.vcxproj'">
```

Comment:
Makes sense to be able to opt into the double-eval even for a vcxproj if you really want to.